### PR TITLE
GPU: support BTC account metrics without collateral

### PR DESCRIPTION
**Pull request by enarjord**
## Summary
- add BTC-denominated account-equity proxy metrics for Apple MPS when BTC collateral is disabled
- derive BTC daily equity from the existing compact USD surface and canonical BTC/USD data, including candidate-specific liquidation endpoints
- keep USD-only runs unchanged and bind BTC data to checkpoint identity only when BTC scoring or limits request it
- retain fail-closed rejection of positive `backtest.btc_collateral_cap` and document approximation boundaries

## Supported BTC proxy surface
Gain, ADG, MDG, Omega, equity shape, unweighted exposure ratios, peak recovery, per-side exposure-normalized gain/ADG/MDG, and the safe weighted ADG/MDG/Omega and equity-shape variants. Metrics requiring synchronized intraday BTC-equity minima (Sharpe, Sortino, expected shortfall, drawdown, Calmar, and Sterling) and weighted exposure ratios remain fail-closed until Metal retains the required surfaces. Exact Rust remains authoritative through mandatory validation and drift gates.

## Validation
- `python -m pytest -q tests/optimization`
- `python -m pytest -q tests/optimization/test_gpu_mps.py` on physical Apple M3/MPS (679 passed)
- `cargo test --manifest-path passivbot-rust/Cargo.toml --no-default-features` (268 passed)
- `cargo check --manifest-path passivbot-rust/Cargo.toml`
- `python -m compileall -q src tests/optimization`
- end-to-end BTC-scored EMA Anchor GPU smoke using ADG/MDG/Omega: 8 generations, 512 proxy evaluations, 16 exact validations, clean completion

`mkdocs build --strict` reaches the documentation build but aborts on the two pre-existing `release_notes_v8.0.0.md` / `release_notes_v8.1.0.md` links to `../CHANGELOG.md`; this PR introduces no new documentation warnings.

### CODE REVIEW

**Comment by enarjord** [comment]
@codex review

File: src/optimization/gpu/metrics.py [hunk 1]
```diff
@@ -1480,6 +1522,314 @@ def _strategy_eq_recovery_distribution_metrics(out: dict) -> dict:
...
+        "expected_shortfall_1pct_btc": expected_shortfall,
+        "gain_btc": gain,
+        "mdg_btc": mdg,
+        "omega_ratio_btc": omega,
+        "sharpe_ratio_btc": sharpe,
+        "sortino_ratio_btc": sortino,
+        "sterling_ratio_btc": sterling,
+    }
+    for name in tuple(values):
+        values[name] = torch.where(has_fill, values[name], zeros)
```
[the remark this answers is not in the document]
Comment (enarjord) [reply src/optimization/gpu/metrics.py:1650]: Addressed in 34e05d6ea: the BTC drawdown metrics are no longer in the supported Metal proxy surface, and a regression test verifies the canonical names remain excluded. No-fill candidates therefore cannot receive the misleading zero drawdown proxy value.

File: src/optimization/gpu/metrics.py [hunk 2]
```diff
@@ -45,6 +45,47 @@
...
+    "drawdown_worst_mean_1pct_btc",
+    "equity_choppiness_btc",
+    "equity_choppiness_w_btc",
+    "equity_jerkiness_btc",
+    "equity_jerkiness_w_btc",
+    "expected_shortfall_1pct_btc",
+    "exponential_fit_error_btc",
+    "exponential_fit_error_w_btc",
+    "exposure_mean_ratio_btc",
+    "exposure_ratio_btc",
```
[the remark this answers is not in the document]
Comment (enarjord) [reply src/optimization/gpu/metrics.py:58]: Addressed in 34e05d6ea: the user docs, changelog, and PR description now explicitly exclude weighted BTC exposure ratios. They remain fail-closed until the proxy retains the suffix-local exposure series needed to reproduce Rust semantics.

File: src/optimization/gpu/metrics.py [hunk 3]
```diff
@@ -1480,6 +1522,314 @@ def _strategy_eq_recovery_distribution_metrics(out: dict) -> dict:
...
+    max_dd = day_max_dd.max(dim=1).values
+    worst_one_pct = _mean_worst_one_pct_largest(day_max_dd, active)
+    gain, adg = _smoothed_gain_adg(day_end_btc, active)
+    daily_changes, change_mask = _pct_change(day_end_btc, active)
+    mdg = _masked_median(daily_changes, change_mask)
+    omega = _omega_ratio(daily_changes, change_mask)
+    daily_min_changes, min_change_mask = _pct_change(day_min_btc, active)
+    sharpe, sortino = _sharpe_sortino(daily_min_changes, min_change_mask, adg)
+    expected_shortfall = _mean_worst_one_pct_abs(
+        daily_min_changes, min_change_mask
```
[the remark this answers is not in the document]
Comment (enarjord) [reply src/optimization/gpu/metrics.py]: Addressed in 34e05d6ea: all BTC metrics that depend on synchronized intraday BTC-equity minima were removed from the proxy surface (Sharpe, Sortino, expected shortfall, drawdown, Calmar, and Sterling, including weighted forms). The implementation now retains only safe day-end-derived metrics and fails closed for the removed names; the docs and tests state that boundary.

**Comment by enarjord** [comment]
@codex review

Please review exact head 06224573da52985aa7a33eef7ca5d655fd391cda after the addressed findings.
